src/pairwise_dist/_pairwise_dist.pyx

```diff
@@ -0,0 +1,58 @@
+# cython: language_level=3
+# cython: cdivision=True
+# cython: boundscheck=False
+# cython: wraparound=False
+# cython: profile=False
+# cython: linetrace=False
+# cython: binding=False
+# distutils: define_macros=CYTHON_TRACE_NOGIL=0
+
+cimport numpy as np
+import numpy as np
+from libc.math cimport fabs
+from cython.parallel cimport prange
+from cython cimport floating, integral
+cimport cython
+
+cdef floating _l1_distance(
+    floating[:, ::1] X_a,
+    int i,
+    floating[:, ::1] X_b,
+    int j,
+    integral n_features,
+) nogil:
+    cdef:
+        int k
+        floating dist = 0
+    
+    for k in range(n_features):
+        dist += fabs(X_a[i][k] - X_b[j][k])
+        
+    return dist
+
+cdef void _pairwise_dist(
+    floating[:, ::1] X_a, # IN
+    floating[:, ::1] X_b, # IN
+    floating[:, ::1] distances, # OUT
+) nogil:
+    cdef:
+        int i, j
+        int n_rows_X_a = X_a.shape[0]
+        int n_rows_X_b = X_b.shape[0]
+        int n_features = X_a.shape[1]
+        
+    for i in prange(n_rows_X_a, nogil=True):
+        for j in range(n_rows_X_b):
+            distances[i, j] = _l1_distance(X_a, i, X_b, j, n_features)
```

Comment:
You can inspect in another PR the parallelization strategy when `n_rows_X_b >> n_rows_X_a`.